Sources/Client.swift

```diff
@@ -0,0 +1,334 @@
+    ///   - account: String representing Ethereum address.
+    ///   - message: String representin Data to sign.
+    ///   - completion: Response with string representing signature, or error.
+    /// - Throws: client error.
+    public func eth_signTypedData(url: WCURL,
+                                  account: String,
+                                  message: String,
+                                  completion: @escaping RequestResponse) throws {
+        try sign(url: url, method: "eth_signTypedData", param1: account, param2: message, completion: completion)
+    }
+
+    private func sign(url: WCURL,
+                      method: String,
+                      param1: String,
+                      param2: String,
+                      completion: @escaping RequestResponse) throws {
+        let payload = JSONRPC_2_0.Request(method: method,
+                                          params: .positional([.string(param1), .string(param2)]),
+                                          id: .string(UUID().uuidString))
+        let request = Request(payload: payload, url: url)
+        try send(request, completion: completion)
+    }
+
+    /// https://docs.walletconnect.org/json-rpc/ethereum#eth_sendtransaction
+    /// Request to send a transaction.
+    ///
+    /// - Parameters:
+    ///   - url: WalletConnect url object.
+    ///   - transaction: Transaction object.
+    ///   - completion: Response with string representing transaction hash, or error.
+    /// - Throws: client error.
+    public func eth_sendTransaction(url: WCURL,
+                                    transaction: Transaction,
+                                    completion: @escaping RequestResponse) throws {
+        try handleTransaction(url: url, method: "eth_sendTransaction", transaction: transaction, completion: completion)
+    }
+
+    /// https://docs.walletconnect.org/json-rpc/ethereum#eth_signtransaction
+    /// Request to sign a transaction.
+    ///
+    /// - Parameters:
+    ///   - url: WalletConnect url object.
+    ///   - transaction: Transaction object.
+    ///   - completion: Response with string representing transaction signature, or error.
+    /// - Throws: client error.
+    public func eth_signTransaction(url: WCURL,
+                                    transaction: Transaction,
+                                    completion: @escaping RequestResponse) throws {
+        try handleTransaction(url: url, method: "eth_signTransaction", transaction: transaction, completion: completion)
+    }
+
+    private func handleTransaction(url: WCURL,
+                                   method: String,
+                                   transaction: Transaction,
+                                   completion: @escaping RequestResponse) throws {
+        let requestParamsData = try JSONEncoder().encode([transaction])
+        let requestParams = try JSONDecoder().decode(JSONRPC_2_0.Request.Params.self, from: requestParamsData)
```

Comment:
Not sure if I like this pattern of encoding the transaction as JSON and then decoding it as params object. I would explicitly map the transaction to the request's params. Maybe it is just a feeling.